packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -151,6 +151,11 @@ export function getSchemaForType(
     }
     return builtinType;
   }
+
+  if (type.kind === "ModelProperty") {
```

Comment:
interesting the model property could also be used as a type also. 
it seems this is not allowed in typescript: https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgILIN4FgBQz-JwBcyAzmFKAOYDcuAvrrqJLIigEKa4HIImoAdHDo56QA.

Reply:
in typespec, it's a valid syntax.